DevProxy/Proxy/ProxyEngine.cs

```diff
@@ -615,21 +615,44 @@ private static void ToggleSystemProxy(ToggleSystemProxyAction toggle, string? ip
         process.WaitForExit();
     }
 
-    private static bool IsFirstRun()
+    internal static class HasRunFlag
     {
-        var firstRunFilePath = Path.Combine(ProxyUtils.AppFolder!, ".hasrun");
-        if (File.Exists(firstRunFilePath))
+        private static readonly string filename = Path.Combine(ProxyUtils.AppFolder!, ".hasrun");
+
+        public static bool IsFirstRun()
         {
-            return false;
+            if (File.Exists(filename))
+            {
+                return false;
+            }
+
+            return CreateFlag();
         }
 
-        try
+        private static bool CreateFlag()

```

Comment:
Since we already have `Flag` in the class name, let's avoid repeating it here and simply call it `Create`.